scripts/origin.js

```diff
@@ -0,0 +1,42 @@
+---
+layout: null
+---
+(function() {
+    function addOrUpdateQueryParam(url, param, value) {
+        if (!url || typeof url !== 'string') return url;
+        
+        var urlObj = new URL(url, window.location.origin);
+        urlObj.searchParams.set(param, value);
+        return urlObj.href;
+    }
+    
+    
+    function shouldModifyUrl(pathname) {
+        if (!pathname || typeof pathname !== 'string') return false;
```

Comment:
I think we need to restrict this to URLS on the DDG domain?

Otherwise I think this is going to match any root path on any domain?